proposals/log_output_v2.md

```diff
@@ -0,0 +1,337 @@
+
+#### Debug
+
+Messages that are useful for debugging what went wrong, but not as useful (or convoluting) as user-facing information. These would be helpful to include in, for example, a bug report.
+
+#### Warn/Error
+
+These messages will be displayed by default, but as the warning/error format of the specific Maven/Gradle logger implementations. Jib should not explicitly log error messages, but rather throw those messages as part of exceptions.
+
+### Information to include
+
+The information to include, in the order in which they should appear are:
+
+1. (info) Where inputs are inferred from, including
+    1. The main class
+1. (info) Files that went into each layer
+    1. Classes, resources, dependencies
+1. (lifecycle/info) Important build steps
+    1. Show what’s skipped?
+    1. Asynchronous steps (the order may vary between runs)
+        - [lifecycle] Retrieving credentials for \<base image>...
+            - [info] Credentials for \<target image> found via (docker-credential-*/Docker config/Maven settings) in \<### ms>
+        - [lifecycle] Retrieving credentials for \<target image>...
+            - [info] Credentials for \<target image> found via (docker-credential-*/Docker config/Maven settings) in \<### ms>
+        - [lifecycle] Getting base image \<base image>…
+            - [info] Fetching manifest for \<base image>...
+            - [info] Fetched manifest for \<base image> in \<### ms>
+            - [info] Fetching layer \<digest> of \<base image>...
+            - [info] Fetched layer \<digest> of \<base image> in \<### ms>
+            - [info] Pulled base image in \<### ms>
+        - [lifecycle] Building dependencies layer...
+            - [info] Built and cached dependencies layer \<digest> in \<### ms>
+            - [info] Pushing dependencies layer \<digest>...
+            - [info] Pushed dependencies layer \<digest> in \<### ms>
+        - [lifecycle] Building resources layer
+            - Same as for dependencies
+        - [lifecycle] Building classes layer
+            - Same as for dependencies
+        - [lifecycle] Finalizing…
+            - [info] Pushing container configuration \<digest>...
+            - [info] Pushed container configuration \<digest> in \<### ms>
+            - [info] Pushing manifest...
+            - [info] Pushed manifest in \<### ms>
+        - [lifecycle] Build complete in \<### s>
+1. Finalized information
+    1. Container entrypoint set to \<entrypoint>
+    1. Built and pushed image as \<target image>
+
+### Example log
+
+At lifecycle level:
+
+```
+[lifecycle] Containerizing application to gcr.io/my-gcp-project/my-app...
+[lifecycle]
+[lifecycle] Retrieving credentials for gcr.io/distroless/java…
+[lifecycle] Retrieving credentials for gcr.io/my-gcp-project/my-app...
+[lifecycle] Getting base image gcr.io/distroless/java…
+[lifecycle] Building dependencies layer...
+[lifecycle] Building resources layer...
+[lifecycle] Building classes layer...
+[lifecycle] Finalizing…
```

Comment:
Finalizing seems like it doesn't do much here, we are already saying `build completed` on the next line. I think we should either make it more descriptive or remove it.

Perhaps `pushing layers?`

Reply:
This part is pushing the container configuration and image manifest. I was thinking that these would be more technical detail than needed. Maybe like "Committing image..." or "Configuring metadata..." might work?

Reply:
I like "configuring metadata", but I also like the info that the build process is finishing up. Maybe something like "Pushing final image configuration..." or just "Finalizing configuration..." could work.

Reply:
How about `Finalizing container image...`? The reason I added this line is so that the user isn't misled to think that the classes layer takes the longest to build.

Reply:
Or maybe just "Pushing..."? Since after "Building", "Pushing" would be what is expected.

Reply:
@loosebazooka